src/types/helperTypes.d.ts

```diff
@@ -93,6 +93,10 @@ export type ConfiguredData<Name extends string> = Name extends keyof DataConfig
 
 export type ConfiguredSource<Name extends string> = Name extends keyof SourceConfig ? SourceConfig[Name] : {};
 
+export type ConfiguredRollData<T extends Document<any, any>, Name extends string> = Name extends keyof GetRollDataConfig
+  ? GetRollDataConfig[Name]
+  : T['data']['data'];
+
```

Comment:
This type slightly breaks convention by taking the document being operated on as the first parameter. The others only take the `Name` parameter. The type could be something like:
```ts
export type ConfiguredRollData<Name extends string> = Name extends keyof GetRollDataConfig
  ? GetRollDataConfig[Name]
  : never;
```

But the call site would have to look like this: `OrDefault<ConfiguredRollData<'Actor' / 'Item>, this['data']['data']>` and `OrDefault` is not yet defined. For complete correctness it would look like this:
```ts
type _Equals<T, U> = T extends U ? (U extends T ? true : false) : false;
type And<T extends boolean, U extends boolean> = _Equals<T, true> extends true
	? _Equals<U, true> extends true
		? true
		: false
	: false;
type IsAny<T> = _Equals<0, 1 | T>;
type Equals<T, U> = And<_Equals<T, U>, _Equals<IsAny<T>, IsAny<U>>>;

export type OrDefault<T, U> = Equals<T, never> extends true ? U : T;
```
Unfortunately this adds a few types that aren't used anywhere else, additionally there would be bikeshedding to do about how `Equals` should handle `any` as well as `OrDefault` only working with `never`.

There's a lot of other solutions that could work here as well but I opted for this for simplicity.